### DISCUSSION TITLE
Performance vs Raspberry Pi version

### DISCUSSION THREAD

**Discussion by ChKK1963** [probonopd/MiniDexed#114] [Q&A] [answered]
Thinking about building this ASAP, is there a substantial performance difference between the Pi 3/4? What is the total number of concurrent voices I can expect?

Does it make any difference for this project if the Raspberry Pi has 2, 4 or 8 GB and of memory?

**Comment by probonopd** [comment, marked as answer]
Welcome @ChKK1963. From my experience:

* Raspberry Pi Zero and 1: Only one Tone Generator (TG), no effects (reverb etc.); "like a DX7"
* All other models: 8 Tone Generators (TG) and effects are available, although using more than 4 concurrently might bring the Pi 2 to its limits
* Pi 4 is best in order not to get stuck USB MIDI keys which can happen (very rarely) under certain circumstances on the other models, see https://github.com/probonopd/MiniDexed/issues/48

The available RAM does not make a difference.

**Reply by ChKK1963** [reply]
Thanks, so a 2GB Pi 4 sounds like the best option.

With 8 TGs in use, what is the number of voices per TG or in total to expect? The original TX816 had 16 per channel …

**Reply by probonopd** [reply]
Same:

https://github.com/probonopd/MiniDexed/blob/7519da5e13ff72a43d05a8a1632d0becb33d457b/src/config.h#L42-L46

If you need more, you could play with that value in the code.

**Reply by ChKK1963** [reply]
👍
